reporters/reporter-default/src/main/java/org/radargun/reporting/html/TestReportDocument.java

```diff
@@ -43,24 +45,39 @@ protected ComparisonChart generateChart(int clusterSize, String operation, Strin
       return chart;
    }
 
-   public void writeTest() throws IOException {
-      writeTag("h1", "Test " + testName);
-
-      for (Map.Entry<String, Map<Report, List<Report.TestResult>>> result : testAggregations.results().entrySet()) {
-         writeTag("h2", result.getKey());
-         writeResult(result.getValue());
-      }
+   public void createTestCharts() throws IOException {
 
       for (String operation : testAggregations.getAllOperations()) {
-         writeTag("h2", operation);
          if (maxClusters > 1 && configuration.separateClusterCharts) {
             for (Integer clusterSize : testAggregations.byClusterSize().keySet()) {
-               createAndWriteCharts(operation, clusterSize);
+               createCharts(operation, clusterSize);
             }
          } else {
-            createAndWriteCharts(operation, 0);
+            createCharts(operation, 0);
          }
-         writeOperation(operation, testAggregations.byReports(), testName);
       }
    }
+
+   // This returns list so that it is compatible with Combined report document
+   @Override
+   public List<TestAggregations> getTestAggregations() {
+      List<TestAggregations> aggregations = new ArrayList<>();
+      aggregations.add(testAggregations);
+      return aggregations;
+   }
+
+   @Override
+   public String singleTestName(int i) {
```

Comment:
Method singleTestName(int i) is also defined in CombinedReportDocument class where the parameter is needed. These two classes share the same template (testReport.ftl) which invokes this method.